src/components/views/auth/InteractiveAuthEntryComponents.js

```diff
@@ -564,6 +600,88 @@ export const MsisdnAuthEntry = createReactClass({
     },
 });
 
+export class SSOAuthEntry extends React.Component {
+    static propTypes = {
+        matrixClient: PropTypes.object.isRequired,
+        authSessionId: PropTypes.string.isRequired,
+        loginType: PropTypes.string.isRequired,
+        submitAuthDict: PropTypes.func.isRequired,
+        errorText: PropTypes.string,
+        onPhaseChange: PropTypes.func.isRequired,
+        continueText: PropTypes.string,
+        continueKind: PropTypes.string,
+        onCancel: PropTypes.func,
+    };
+
+    static LOGIN_TYPE = "m.login.sso";
+    static UNSTABLE_LOGIN_TYPE = "org.matrix.login.sso";
+
+    static PHASE_PREAUTH = 1; // button to start SSO
+    static PHASE_POSTAUTH = 2; // button to confirm SSO completed
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            // We actually send the user through fallback auth so we don't have to
+            // deal with a redirect back to us, losing application context.
+            ssoUrl: props.matrixClient.getFallbackAuthUrl(
+                this.props.loginType,
+                this.props.authSessionId,
+            ),
+            phase: SSOAuthEntry.PHASE_PREAUTH,
+        };
+    }
+
+    componentDidMount(): void {
+        this.props.onPhaseChange(SSOAuthEntry.PHASE_PREAUTH);
+    }
+
+    onStartAuthClick = () => {
+        // Note: We don't use PlatformPeg's startSsoAuth functions because we almost
+        // certainly will need to open the thing in a new tab to avoid loosing application
+        // context.
```

Comment:
What does this do on electron?

Also, having a different mechanism for UIA SSO to login doesn't seem right, especially when the SSO code has been touched relatively recently. Presumably the problem is that that the SSO stuff just causes another navigation within Riot, but we need the current dialog stack to stay open and continue with the auth flow?

Reply:
yea, we need the dialog stack to stay open so we can keep all the references to the functions that need to be called once auth is complete. Login is an exception because it needs to do a bunch of redirects back into the app to get the token from UIA, but here we just need a pass/fail mark.

I haven't tested it explicitly, but the `window.open` should pop into a browser, and we don't need to talk between the frames.

Reply:
OK - presumably this won't be a great experience since the user will just be told to go back & find their matrix client, so as long as we make sure we schedule in time to go back & fix this properly.

Reply:
We aren't planning on fixing this properly, as this is proper for the time being.

Reply:
Ok :/

Reply:
sorry :( 

UIA needs a bunch more design time that we don't have to make it work nicely.